nix/grafana/default.nix

```diff
@@ -0,0 +1,105 @@
+{ pkgs, lib, name, config, ... }:
+let
+  inherit (lib) types;
+  iniFormat = pkgs.formats.ini { };
+in
+{
+  options = {
+    description = ''
+      Configure grafana.
+    '';
+    enable = lib.mkEnableOption name;
+
+    package = lib.mkPackageOption pkgs "grafana" { };
+
+    http_port = lib.mkOption {
+      type = types.int;
+      description = "Which port to run grafana on.";
+      default = 3000;
+    };
+
+    domain = lib.mkOption {
+      type = types.str;
+      description = "The public facing domain name used to access grafana from a browser.";
+      default = "localhost";
+    };
+
+    protocol = lib.mkOption {
+      type = types.str;
+      description = "Protocol (http, https, h2, socket).";
+      default = "http";
+    };
+
+    root_url = lib.mkOption {
+      type = types.str;
+      description = "The full public facing url.";
+      default = "${config.protocol}://${config.domain}:${builtins.toString config.http_port}";
+    };
```

Comment:
If we don't expect the user to set this, you want to add `readOnly = true;` here.

Reply:
grafana actually allows to give a `root_url` with subpath. in that case, user may set this value.

ref: https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#root_url

Reply:
Will the user simply not set `extraConfig.server.root_url` instead? This additional option might not be needed. I think the [latest impl. of readiness_probe](https://github.com/juspay/services-flake/pull/108/commits/98811170929b85415a9e2c42b0e03a0f402435fd) looks good as is